truss-chains/truss_chains/framework.py

```diff
@@ -532,62 +531,6 @@ def _get_generic_class_type(var):
     return origin if origin is not None else var
 
 
-def _validate_dependency_arg(
```

Comment:
This was moved to the `_InitValidator` below, since it seemed to match with other class functions to validate self / context args (and nobody else used this outside the `_InitValidator`)